lightning/src/ln/blinded_payment_tests.rs

```diff
@@ -1183,3 +1183,84 @@ fn conditionally_round_fwd_amt() {
 	let expected_fee = pass_claimed_payment_along_route(args);
 	expect_payment_sent(&nodes[0], payment_preimage, Some(Some(expected_fee)), true, true);
 }
+
+#[test]
+fn blinded_keysend() {
+	let mut mpp_keysend_config = test_default_channel_config();
+	mpp_keysend_config.accept_mpp_keysend = true;
```

Comment:
That seems ok to me. But why does it need to be set `true` in this test if `blinded_mpp_keysend` tests MPP?

Reply:
`accept_mpp_keysend` means that users opt into potentially breaking compat with < 0.0.116, which can happen if `PendingHTLCRouting::ReceiveKeysend::payment_data` is set. Because the payment secret will always be present in keysends to blinded paths, `::payment_data` will always be `Some`, so users need to opt into the potential compat breakage. Does that make sense?

Reply:
> I guess? ISTM we could instead actually just use the `keysend_preimage` - only the sender has that, so no need to worry about the secret anywhere. In general I kinda anticipated no one would support reading a payment secret in a blinded-path-receive HTLC, but certainly we could require it 🤷

I'm confused, we currently require payment secret to be present in all blinded path receives (so we can do stateless inbound payment verification). So the user isn't providing the payment secret, we're providing it back to ourselves in the blinded path.